meterfs/test_meterfs_allocate.c

```diff
@@ -0,0 +1,119 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Meterfs allocating tests group
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Tomasz Korniluk
+ *
+ *
+ * %LICENSE%
+ */
+
+#include "common.h"
+
+TEST_GROUP(meterfs_allocate);
+
+
+TEST_SETUP(meterfs_allocate)
+{
+}
+
+
+TEST_TEAR_DOWN(meterfs_allocate)
+{
+	TEST_ASSERT_EQUAL(0, file_eraseAll());
+}
+
+
+/* NOTE: not valid for hw target .Now there is no way to obtain/parametrize flash size in test */
+/* test case of allocating file bigger than flash size */
+TEST(meterfs_allocate, big_file)
+{
+	file_info_t info;
+	uint32_t flashsz = 4 * 1024 * 1024;
+	uint32_t sectorsz = 4096;
+
+	common_fileInfoFill(&info, ((flashsz + sectorsz) / sectorsz) + 1, flashsz + sectorsz, 20, 0);
+
+	TEST_ASSERT_EQUAL(-EINVAL, file_allocate("file0", info.sectors, info.filesz, info.recordsz));
+}
+
+
+/* test case of allocating more files than allowed */
+TEST(meterfs_allocate, many_files)
+{
+	int i;
+	char fileName[9];
+
+	for (i = 0; i < (255 + 10); ++i) {
+		sprintf(fileName, "file%d", i);
```

Comment:
Maybe instead of `sprintf()` safe `snprintf()`.

Reply:
Done.